app/src/androidTest/java/org/dharmaseed/android/TalkRepositoryTest.java

```diff
@@ -0,0 +1,215 @@
+package org.dharmaseed.android;
+
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+import android.database.sqlite.SQLiteOpenHelper;
+import android.support.test.InstrumentationRegistry;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.*;
+
+public class TalkRepositoryTest
+{
+
+    private static SQLiteOpenHelper dbManager;
+    private TalkRepository talkRepository;
+
+    @BeforeClass
+    public static void setUp() throws Exception
+    {
+        createDB();
+    }
+
+    private static void createDB() throws Exception
+    {
+        dbManager = new MockDBManager(InstrumentationRegistry.getTargetContext());
+        SQLiteDatabase db = dbManager.getWritableDatabase();
+        InputStream fis = InstrumentationRegistry.getContext().getAssets().open("inserts.txt");
+        String line = "";
+        int ch;
+        // run every statement from "inserts.txt"
+        while ((ch = fis.read()) != -1)
+        {
+            if (ch < 8 || ch > 'z') continue;
+            line += (char) ch;
+            // execute the statement once we hit a semicolon followed by a newline
+            if (ch == ';' && fis.read() == '\n')
+            {
+                db.execSQL(line);
+                line = "";
+            }
+        }
+    }
+
+    @AfterClass
+    public static void tearDown() throws Exception
+    {
+        dbManager.close();
+    }
+
+    @Test
+    public void selectIds_isCorrect()
+    {
+        talkRepository = new TalkRepository(dbManager);
+
+        List<String> columns = new ArrayList<>();
+        columns.add(DBManager.C.Talk.TABLE_NAME + "." + DBManager.C.Talk.ID);
+        Cursor actualCursor = talkRepository.getTalks(columns, null, false, false);
```

Comment:
Thanks a lot for adding these tests, I think they'll be super handy to have!  I have a few questions:

1) It seems like this test (and also `selectIdsTeacherIds_isCorrect` below) are a subset of `getAllTalks_isCorrect`, in the sense that `getAllTalks_isCorrect` validates all columns of the talks table, whereas the former two tests just test one or two columns.  My initial reaction here was to wonder whether it would thus be possible to eliminate the former two without diminishing the test coverage, especially since `assertCursors` already nicely reports which columns fail.  But I'm happy to keep them if you think there's extra value provided by the more specialized tests. :)

Related to that,
2) I'm wondering whether the changes I made to `getTalks` in #60 [here](https://github.com/dharmaseed/dharmaseed-android/pull/60/files#diff-c96af1a84df0db1d88f4ffedc6459a50L49), where I removed the `columns` argument and made the function return the same columns every time, might have created a logical conflict with these tests.  In particular, it won't be possible to test only a subset of columns like we do here.  I'm happy to make any required changes to #60 once we merge this (if you haven't already; I know you said you've already fixed up some things), but if you have any thoughts on that let me know! :)

Reply:
1. You're right that they are a subset in that they all test the same function but with different columns. But I think this is useful because it tests how the `getTalks()` function reacts to a) no columns provided; b) one column provided; and c) multiple columns provided. And yes, I do think there is extra value provided by having these more specialized tests.

2. Yes, there will be a conflict with the change you made in #60. I noticed this too, but I was waiting to post most of my comments on your PR until after we merged this PR. I will still post my comments about the change you made to `getTalks()`, but in short I don't think we should remove that param. I'll elaborate further in your PR.

Reply:
Cool, that all sounds good to me. :)  In that case, I'll approve and merge this as soon as you add the extra tests for starred and downloaded filters, and then we can discuss appropriate next steps in #60.

Reply:
Awesome! I just added a bunch of tests.

Reply:
Cool, the new tests look great!